internal/provider/cluster/cluster.go

```diff
@@ -39,6 +39,8 @@ func ResourceCluster() *schema.Resource {
 			Delete: schema.DefaultTimeout(clusterDeleteTimeout),
 		},
 
+		SchemaVersion: 1,
```

Comment:
You need to provide also an upgrade func to migrate state files.

Otherwise if there is no need to migrate state file (because existing gcp clusters will continue to work with this change), there is no need to increase the schema version.

Reply:
OK, I've added migration inspired by what has been done with VPC peering.